Sources/Plasma/PubUtilLib/plNetGameLib/Private/plNglFile.cpp

```diff
@@ -1112,7 +1120,8 @@ bool DownloadRequestTrans::Send () {
         return false;
 
     Cli2File_FileDownloadRequest filedownloadReq;
-    StrCopy(filedownloadReq.filename, m_filename.WideString().data(), std::size(filedownloadReq.filename));
+    const ST::utf16_buffer buffer = m_filename.AsString().to_utf16();
+    StrCopy(filedownloadReq.filename, buffer.data(), std::size(filedownloadReq.filename));
```

Comment:
Why not the std::char_traits copy method?

Reply:
Because `st::char_traits<T>::copy` is more equivalent to `memcpy()`, wheras `StrCopy()` is closer to `strncpy()`.  Or more specifically, `char_traits` copies exactly the specified number of characters (regardless of content) whereas `StrCopy` copies up to either the first `'\0'` or the specified maximum.